src/test/java/pro/trevor/tankgame/util/TankBuilder.java

```diff
@@ -0,0 +1,46 @@
+package pro.trevor.tankgame.util;
+
+import org.json.JSONObject;
+
+import pro.trevor.tankgame.rule.impl.version3.Tank;
+import pro.trevor.tankgame.rule.impl.version3.TankAttribute;
+import pro.trevor.tankgame.state.board.Position;
+import pro.trevor.tankgame.state.board.attribute.IAttribute;
+import pro.trevor.tankgame.state.board.unit.GenericTank;
+
+public class TankBuilder<T extends GenericTank<E>, E extends Enum<E> & IAttribute> 
+{
+    private final T tank;
+    public TankBuilder(T tank) 
+    {
+        this.tank = tank;
+    }
+
+    public TankBuilder<T, E> withAttribute(E attribute, Object value)
+    {
+        tank.set(attribute, value);
+        return this;
+    }
+
+    public TankBuilder<T, E> atPosition(Position p)
+    {
+        tank.setPosition(p);
+        return this;
+    }
+
+    public T finish()
```

Comment:
I would name this `build`

Reply:
Right, but I wouldn't, so I didn't.

Reply:
It starts with "build...()", so ending with something different just makes sense.